### PR TITLE
975 improve launch sequence

### PR DESCRIPTION
**Pull request by BPerlakiH**
Fixes: #975

In combination with the [library loading improvements](https://github.com/kiwix/kiwix-apple/pull/974) gives a way better UX:

https://github.com/user-attachments/assets/23f8b9bf-ca62-447c-abfd-a1c4914bdd43

### CODE REVIEW

**Review by rgaudin — APPROVED** [review]
Please provide more details about what this does.
AFAIU, we know have a spashscreen with centered kiwix-logo in lieu of a blank screen when starting the app.
I'm not sure what happens during this time but on my very old an slow iphone, it's visible but not noticeable (compared to other apps).

**Comment by kelson42** [comment]
My remarks:
* Please make the visual 50% bigger
* Make the splash screen dark-mode compatible (with default OS colorsets)
* @rgaudin Please share the link to the official visuals for both dark and light modse
* The logo should be displayed the whole time at the exact same place, and "Loadding... " should be displayed in place of the buttons as long as they are not available.

**Comment by BPerlakiH** [comment]
Please see the updated sequences on all type of devices, we have the combination of portrait / landscape and light / dark modes:

----------------
# IPHONE

https://github.com/user-attachments/assets/c2956846-d84b-4d4a-9b96-b53bb0a0f1a3

https://github.com/user-attachments/assets/6ec11d88-6b8a-4943-bd5f-eb0c6fd73cdf

----------------
# IPAD

https://github.com/user-attachments/assets/bf54d987-74ff-4f15-a558-21a52be6f874


https://github.com/user-attachments/assets/2addf569-f2ac-4a25-82a8-0dad28f5c9ac


https://github.com/user-attachments/assets/4e47fac1-a969-44a5-a52c-e6bd139707ba


https://github.com/user-attachments/assets/56cd3b81-864a-464f-9711-3d9e97e4ff74

----------------
# MAC

https://github.com/user-attachments/assets/0d23f05e-d458-4237-bb45-a874816120c6


https://github.com/user-attachments/assets/0e071911-7eb4-4e61-8bc7-d886db1a87bc

**Comment by BPerlakiH** [comment]
We have a slight movement of the logo from the splash/landing screen to the loading view on iPad, due to the fact that the welcome screen is shifted by the side navigation panel. Compare with collapsed / expanded navigation bar the "welcome" screen:
ipad_2
ipad_1

**Comment by BPerlakiH** [comment]
Note: on macOS the launch/splash screen is not supported, so we only have the loading sequence next to the navigation bar.

**Comment by BPerlakiH** [comment]
Looks OK now, I've tested with the DWDS app.

**Comment by kelson42** [comment]
I guess we need real logo now.

**Comment by rgaudin** [comment]
https://github.com/kiwix/overview/tree/main/branding

**Comment by BPerlakiH** [comment]
@kelson42 do we want to use the vertical or the horizontal logo?

**Comment by kelson42** [comment]
@BPerlakiH Sorry for not reacting to this issue, somehow I missed it on my radar. I believe the vertical is better, in particular for iPhones.

**Comment by BPerlakiH** [comment]
I have updated the logo, and fix a minor resizing issue. Retested on iPhone, iPad and macOS:

https://github.com/user-attachments/assets/96640eca-33f1-4606-84e7-16780874ad4b


https://github.com/user-attachments/assets/2573f0d2-2283-451b-b6e7-daaea0756b6a



https://github.com/user-attachments/assets/660d81a5-1d15-460d-aee9-aff36a3478b0

**Comment by kelson42** [comment]
@BPerlakiH The "Loading..." step has disappeared? why?

**Comment by kelson42** [comment]
In term of UI, many things "disturb" me:
* The separation line, I would remove it
* The fact that the two buttons are on the same ligne, istead of been stack. Stacking the button would allow to have the buttons the same wifth than the logo and properly aligned.
* spaces between bird, kiwix letter and buttons would better be the same
* "Fetching..." is very unclear, maybe "Fetching library..." would be better/clearer.

**Comment by BPerlakiH** [comment]
@kelson42 this is the "loading" state (sometimes the spinner is not visible in QuickTime, as the player controls are on the same level):
Simulator Screenshot - iPhone 15 Pro Max - 2024-09-22 at 00 09 15

**Comment by BPerlakiH** [comment]
These are the macOS ones:

https://github.com/user-attachments/assets/ca9c0561-2943-47d5-bd73-68412e14cc7f


https://github.com/user-attachments/assets/2851df24-7bfd-448a-ad44-1b142f8bc54a

**Comment by BPerlakiH** [comment]
I agree, will tweak the UI.
Also on the same note, things that bothers you (and me), I separated one of those into a new issue, around the naming (not just the button), please see: https://github.com/kiwix/kiwix-apple/issues/990

**Comment by kelson42** [comment]
@BPerlakiH Agree, after thinking "twice" about it, please remove the sidebar from thensplash screen. That way is clearer.

File: App/App_macOS.swift [hunk 1]
```diff
@@ -110,7 +110,7 @@ struct RootView: View {
     private let tabCloses = NotificationCenter.default.publisher(for: NSWindow.willCloseNotification)
 
     var body: some View {
-        NavigationView {
+        NavigationSplitView {
```
Comment (BPerlakiH) [inline App/App_macOS.swift:113]: In accordance to Apple's recommendation from: 
https://developer.apple.com/documentation/swiftui/migrating-to-new-navigation-types

**Comment by BPerlakiH** [comment]
@kelson42, @rgaudin New recordings of loading sequences: 

## iPad:

https://github.com/user-attachments/assets/482caf88-7e40-4c16-963f-30dab8cac625



------------------

## iPhone:


https://github.com/user-attachments/assets/61a3d875-5370-4e04-97a9-7a4b47a0b373


------------------

## macOS:


https://github.com/user-attachments/assets/e3855209-1f00-474f-89b2-2592d1e2b9c0

**Comment by BPerlakiH** [comment]
And this is the loading sequence, when there's already a local ZIM file on iPad:

https://github.com/user-attachments/assets/fa8ef1d5-e0a0-49f0-8053-b224533e6de9

**Comment by kelson42** [comment]
This is not exactly what we discussed, I will come back to you with a mockup so there is no ambiguity anymore on the requirements.

**Comment by kelson42** [comment]
For all screens, the "block" logo + buttons should be perfectly centered
For the Splash screen:
* In the place of the 2 buttons, perfectly centered in the middle of the place where the two buttons will appear, write "Loading Application..." and the three points should be printed after each other endlessly to show it is progressing)
image
* One we leave the splash screen, we should have the same but displaying in place "Loading application...": "Loading Data..."
* Then finally there is the moment where we display at this place "Fetching Catalog..."
* Finally, once everything is loaded, then we (migh) display the button like following (in red are spaces to remove):
image

Remarks:
* All of this implies that the spinner disappear. The reason is that it seems there is no proper way to have it already running properly at the splash screen level which then create a look & feel disruption. This is why I have replaced it with strings (which at the end might be gifs to be animated.

**Comment by BPerlakiH** [comment]
@kelson42 I have updated it. Please have a look:

https://github.com/user-attachments/assets/8f22576b-b9e9-4789-9ce5-5f89e0b5aaa7

**Comment by kelson42** [comment]
Buttons don't have the proper width and don't display the buttons "in a flash" if at the end you want to display the library.

**Comment by BPerlakiH** [comment]
By matching the button width with the logo width, in some languages (eg French) the button label can run over that, and will look:
Simulator Screenshot - iPad Pro (12 9-inch) (2nd generation) - 2024-10-03 at 08 54 04 Medium

**Comment by BPerlakiH** [comment]
We can make the logo bigger though...
I will need to dig deeper into this, since hard coding the logo size, isn't a good way either, as for custom apps the logo has a different aspect ratio.

**Comment by kelson42** [comment]
> We can make the logo bigger though...

Yes, this is still too small to my taste.

**Comment by kelson42** [comment]
@BPerlakiH Don't merge main to the feature branch!

**Pull request closed without merging**